oauth2/handler.go

```diff
@@ -156,9 +156,23 @@ func (h *handler) serveAuthorizationRequest(
 	// redirect_uri
 	if redir, ok := query["redirect_uri"]; ok {
 		for _, r := range client.Redirects {
-			if redir[0] == r {
-				authzState.RedirectURI = r
-				break
+			lastChar := r[len(r)-1:]
+			if lastChar == "*" {
+				length := len(r) - 1
+				if len(redir[0]) >= length {
+					// do partial string match up to the '*' character, e.g.
+					// https://host/redirect/to/anywhere will match https://host/redirect/*
+					if r[:length] == redir[0][:length] {
+						authzState.RedirectURI = redir[0]
+						break
+					}
+				}
+			} else {
+				// do an exact string match
+				if redir[0] == r {
+					authzState.RedirectURI = r
+					break
```

Comment:
Je hebt twee break statements voor de for loop, dat maakt het lastiger om de logica te volgen en zeker te weten of er geen infinite loop mogelijk is. 

Een duidelijkere approach zou zoiets zijn:

```
var targetRedirect = determineRedirect(redirectRequestedURL, allowedRedirectPattern)
authzState.RedirectURI = targetRedirect;
break

func determineRedirect(string redirectRequestedURL, allowedRedirectPattern) string {
    
   if (lastchar == *) && strings.hasPrefix {
      //in case the requestRedirectURL matches the allowed pattern, then return requested
      return redirectRequestedURL;
    }
    // no match on the pattern, redirect to the configured url
    return allowedRedirectPattern;
}

}
```

Reply:
Gracefully ignored as I want to change as little as I can. Authz is almost EOL and is planned to be replaced by Keybloak. Also, there is 0 risk for an infinite loop. I trust that the changes I made to increase the readability suffice.